bark/models/behavior/idm/base_idm.cpp

```diff
@@ -117,7 +117,14 @@ double BaseIDM::CalcNetDistance(
   const State leading_state = leading_agent->GetCurrentState();
   const float other_velocity = leading_state(StateDefinition::VEL_POSITION);
 
-  FrenetPosition frenet_leading = leading_agent->CurrentFrenetPosition();
+  // we need to use the lane corridor of the ego agent to be able to compare the
+  // frenet values
+  const auto& lane_corridor =
```

Comment:
Another idea would be to pass the observed_world& to `CalcNetDistance(...)`, then you could use
`const auto& lane_corridor = observed_world.GetLaneCorridor()` and `const auto& ego_agent = observed_world.GetEgoAgent()`

Reply:
The idea and implementation itself makes sense!